src/Enum/JobStatus.php

```diff
@@ -17,7 +17,7 @@ class JobStatus
     protected function __construct(int $status)
     {
         if (!in_array($status, $this->available(), true)) {
-            throw new InvalidArgumentException('Invalid status provided');
+            throw new InvalidStatusException();
```

Comment:
I think the status itself would be very helpful to have in both exception message and exception object.